includes/libraries/tracks/class-tracks-client.php

```diff
@@ -0,0 +1,176 @@
+			$event = new Tracks_Event( $event );
+		}
+		if ( is_wp_error( $event ) ) {
+			return $event;
+		}
+
+		$pixel = $event->build_pixel_url( $event );
+
+		if ( ! $pixel ) {
+			return new WP_Error( 'invalid_pixel', 'cannot generate tracks pixel for given input', 400 );
+		}
+
+		return self::record_pixel( $pixel );
+	}
+
+	/**
+	 * Synchronously request the pixel.
+	 *
+	 * @param string $pixel pixel url and query string.
+	 * @return mixed An error or true if successful.
+	 */
+	public static function record_pixel( $pixel ) {
+		// Add the Request Timestamp and URL terminator just before the HTTP request.
+		$pixel .= '&_rt=' . self::build_timestamp() . '&_=_';
+
+		$response = wp_remote_get(
+			$pixel,
+			array(
+				'blocking'    => true, // The default, but being explicit here.
+				'timeout'     => 1,
+				'redirection' => 2,
+				'httpversion' => '1.1',
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$code = isset( $response['response']['code'] ) ? $response['response']['code'] : 0;
+
+		if ( 200 !== $code ) {
+			return new WP_Error( 'request_failed', 'Tracks pixel request failed', $code );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Build an event and return its tracking URL
+	 *
+	 * @deprecated          Call the `build_pixel_url` method on a Tracks_Event object instead.
+	 * @param  array $event Event keys and values.
+	 * @return string       URL of a tracking pixel.
+	 */
+	public static function build_pixel_url( $event ) {
+		$_event = new Tracks_Event( $event );
+		return $_event->build_pixel_url();
+	}
+
+	/**
+	 * Create a timestap representing milliseconds since 1970-01-01
+	 *
+	 * @return string
+	 */
+	public static function build_timestamp() {
+		$ts = round( microtime( true ) * 1000 );
+		return number_format( $ts, 0, '', '' );
+	}
+
+	/**
+	 * Grabs the user's anon id from cookies, or generates and sets a new one
+	 *
+	 * @return string An anon id for the user
+	 */
+	public static function get_anon_id() {
+		static $anon_id = null;
+
+		if ( ! isset( $anon_id ) ) {
+
+			// Did the browser send us a cookie?
+			if ( isset( $_COOKIE['tk_ai'] ) && preg_match( '#^[A-Za-z0-9+/=]{24}$#', $_COOKIE['tk_ai'] ) ) {
+				$anon_id = $_COOKIE['tk_ai'];
+			} else {
+
+				$binary = '';
+
+				// Generate a new anonId and try to save it in the browser's cookies
+				// Note that base64-encoding an 18 character string generates a 24-character anon id.
+				for ( $i = 0; $i < 18; ++$i ) {
+					$binary .= chr( wp_rand( 0, 255 ) );
+				}
+
+				$anon_id = 'jetpack:' . base64_encode( $binary );
+
+				if ( ! headers_sent()
+					&& ! ( defined( 'REST_REQUEST' ) && REST_REQUEST )
+					&& ! ( defined( 'XMLRPC_REQUEST' ) && XMLRPC_REQUEST )
+				) {
+					setcookie( 'tk_ai', $anon_id );
+				}
+			}
+		}
+
+		return $anon_id;
+	}
+
+	/**
+	 * Gets the WordPress.com user's Tracks identity, if connected.
+	 *
+	 * @return array|bool
+	 */
+	public static function get_connected_user_tracks_identity() {
```

Comment:
I don't see this method have been used inside this code, but probably nice to check if `Jetpack` exists first here.